packages/cf.js/test/fixture.ts

```diff
@@ -0,0 +1,49 @@
+import {
+  AppInstanceInfo,
+  AssetType,
+  INodeProvider,
+  Node
+} from "@counterfactual/common-types";
+import { BigNumber } from "ethers/utils";
+import EventEmitter from "eventemitter3";
+
+export class TestNodeProvider implements INodeProvider {
+  public postedMessages: Node.Message[] = [];
+  readonly callbacks: ((message: Node.Message) => void)[] = [];
```

Comment:
@mvanderh It'd be nice to have typed callbacks, I'm pretty sure this signature is used in a bunch of places.

Reply:
You mean define a type alias? I myself prefer repeating the function type explicitly. Function-type typedefs seem opaque to me.

Reply:
Yeah, I guess it's just a matter of style. I think it's more meaningful to say:

```ts
export type NodeProviderCallbackFunction = (message: Node.Message) => void;
```

and then use it as:

```
readonly callbacks: NodeProviderCallbackFunction[];
```

But yeah. I don't really mind :)

Reply:
True. But providing explicit function type means i can tell what the field is in one look. Typealiasing the function type requires me to look at the field type and ALSO at the type alias.